medusa/server/api/v2/cache.py

```diff
@@ -0,0 +1,143 @@
+                    data.get('season'),
+                    data.get('show_name'),
+                    data.get('indexer')]):
+            return self.api_finish(status=400, error="Invalid post body, can't update")
+
+        cache_db_con = db.DBConnection('cache.db')
+        last_changes = cache_db_con.connection.total_changes
+        cache_db_con.action(b'UPDATE scene_exceptions'
+                            b' set indexer = ?'
+                            b', indexer_id = ?'
+                            b', show_name = ?'
+                            b', season = ?'
+                            b', custom = 1'
+                            b' WHERE exception_id = ?',
+                            [data.get('indexer'),
+                             data.get('indexer_id'),
+                             data.get('show_name'),
+                             data.get('season'),
+                             exception_id])
+        if cache_db_con.connection.total_changes - last_changes == 1:
+            return self.api_finish(status=204)
+        return self.api_finish(status=400, error="Could not update.")
+
+    def post(self, *args, **kwargs):
+        """Add a show."""
+        data = json_decode(self.request.body)
+
+        if not all([data.get('indexer_id'),
+                    data.get('season'),
+                    data.get('show_name'),
+                    data.get('indexer')]):
+            return self.api_finish(status=400, error="Invalid post body, can't update")
+
+        cache_db_con = db.DBConnection('cache.db')
+        last_changes = cache_db_con.connection.total_changes
+        cache_db_con.action(b'INSERT INTO scene_exceptions'
+                            b' (indexer, indexer_id, show_name, season, custom) '
+                            b' values (?,?,?,?,1)',
+                            [data.get('indexer'),
+                             data.get('indexer_id'),
+                             data.get('show_name'),
+                             data.get('season')])
+        if cache_db_con.connection.total_changes - last_changes > 0:
+            return self.api_finish(status=204)
+        return self.api_finish(status=400, error="Could not update.")
+
+        return self.api_finish()
+
+    def delete(self, *args, **kwargs):
+        """Delete a show.
+
+        :param exception_id:
+        :type exception_id: str
+        """
+        try:
+            exception_id = int(kwargs.pop('row_id', 'throwme'))
+        except (ValueError, TypeError):
+            return self.api_finish(status=400, error="Require a valid exception row id.")
+
+        cache_db_con = db.DBConnection('cache.db')
+        last_changes = cache_db_con.connection.total_changes
+        cache_db_con.action(b'DELETE FROM scene_exceptions WHERE exception_id = ?', [exception_id])
+        if cache_db_con.connection.total_changes - last_changes > 0:
+            return self.api_finish(status=204, data={'result': 'Deleted {0} row.'
```

Comment:
If you're returning `204 No Content`, better to not have any data returned.

http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#http-status